mycore-ocfl/src/main/java/org/mycore/ocfl/MCROCFLXMLClassificationManager.java

```diff
@@ -73,8 +76,30 @@ protected static char convertMessageToType(String message) throws MCRPersistence
         return MESSAGE_TYPE_MAPPING.get(message);
     }
 
-    protected OcflRepository getRepository() throws ClassCastException {
-        return MCROCFLRepositoryProvider.getRepository(repositoryKey);
+    /**
+     * Initializes the ClassificationManager
+     */
+    public MCROCFLXMLClassificationManager() {
+
+    }
+
+    /**
+    * Initializes the ClassificationManager with the given repositoryKey.
+    * @param repositoryKey the ID for the repository to be used
+    */
+    public MCROCFLXMLClassificationManager(String repositoryKey) {
+        this.repositoryKey = repositoryKey;
+        repository = MCROCFLRepositoryProvider.getRepository(repositoryKey);
+    }
+
+    @MCRPostConstruction
+    @SuppressWarnings("PMD")
+    public void init(String property) {
+        repository = MCROCFLRepositoryProvider.getRepository(repositoryKey);
+    }
```

Comment:
See comment on the MCROCFLXMLUserManager regarding performance impact of initializing it with `new MCROCFL[...]Manager()`.